Upstream retagged 2.3.1 again, and now it does not build:
```
nix-build -A dune_2
these derivations will be built:
  /nix/store/bq120hpmmvai4xqf73xqf3pk4r9bchcn-dune-2.3.1.drv
building '/nix/store/bq120hpmmvai4xqf73xqf3pk4r9bchcn-dune-2.3.1.drv'...
⋮
./.duneboot.exe
Done: 53/647 (jobs: 8)cd _boot && /nix/store/vvhkf305m0ivsxrm645f7lx87x93f7z1-ocaml-4.07.1/bin/ocamlopt.opt -c -g -I +threads fcntl_stubs.c
clang-7: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
Done: 449/647 (jobs: 8)cd _boot && /nix/store/vvhkf305m0ivsxrm645f7lx87x93f7z1-ocaml-4.07.1/bin/ocamlopt.opt -c -g -no-alias-deps -w -49 -pp ./pp/pp.exe -dump-ast -I +threads dune__module_name.ml
File "src/dune/module_name.ml", line 101, characters 10-23:
Error: Unbound value of_string_opt
make: *** [Makefile:30: dune.exe] Error 2
builder for '/nix/store/bq120hpmmvai4xqf73xqf3pk4r9bchcn-dune-2.3.1.drv' failed with exit code 2
error: build of '/nix/store/bq120hpmmvai4xqf73xqf3pk4r9bchcn-dune-2.3.1.drv' failed
```

--- a/pkgs/development/tools/ocaml/dune/2.nix
+++ b/pkgs/development/tools/ocaml/dune/2.nix
@@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
 
   src = fetchurl {
     url = "https://github.com/ocaml/dune/releases/download/${version}/dune-${version}.tbz";
-    sha256 = "166a6lncx0nlmn0kxgx9qkn1dnbphm30vy1wncxcpsqspxhkqfra";
+    sha256 = "1jzm29z58l34kpqll9jcz5dkkhh36lncba1yb32ghknkvyfdvcxj";
   };
 
   buildInputs = [ ocaml findlib ];

